Change close button appearence to be closer to native

ariasuni opened the pull request:
fix #270 
fix #350 

Close button is smaller, closer to the right of the sidebar and less rounded.

Before:
![screenshot_20180331_003536](https://user-images.githubusercontent.com/2446451/38156131-1047037e-347c-11e8-944e-b459e409437c.png)

After:
![screenshot_20180331_003551](https://user-images.githubusercontent.com/2446451/38156102-dc430154-347b-11e8-8f03-ad5f71da44f2.png)

eoger commented:
This looks like a good endeavor to pursue, I think we might as well go all the way and use Firefox's .svg and measurements: https://searchfox.org/mozilla-central/rev/42b34ba1961e37e0d2236deafdd126a0ba21b9ec/toolkit/themes/shared/close-icon.inc.css#8

ariasuni commented:
I updated Photon icons from [here](https://design.firefox.com/icons/viewer/). Note that now, the close icon is black as opposed to some shade of gray in Firefox, but since I do not know well either the CSS of Tab Center Redux nor the CSS properties involved, I’ll probably let that to another person.

ariasuni commented:
I re-did my work to match current version of Tab Center Redux. It seems to me that the current appearance of the close button matches the rest of the extension, while being reasonably similar to the Firefox tab bar.

eoger commented:
Apologies for the delay, I've tried the PR and it's looking good.
However, I think the color of the X should match the one present in the native Firefox tabs.

ariasuni commented:
I tried to use the same code that in the link of [your previous comment](https://github.com/eoger/tabcenter-redux/pull/328#issuecomment-377639682), then made a few searches and tried different ways to change the color used to fill the SVG, with no success.

Basically, [we can’t do it this way](https://developer.mozilla.org/en-US/docs/Web/SVG/Attribute/fill#Using_context-fill), because [we can’t use `-moz-content-properties`](https://developer.mozilla.org/en-US/docs/Web/CSS/-moz-context-properties). So I edited the SVG. :woman_shrugging: I changed the `fill` value of the SVG to the color value used by the light Firefox theme.

Note that in the future, improving the icons for dark theme will simply be a matter of copying each icon and changing the `fill` attribute of the SVG, which is very easy and doesn’t bloat the extension since the SVGs are so tiny.

ariasuni commented:
Are you satisfied with the proposed changes?

eoger commented:
Sorry I'm currently in a place with no computer, I'll try to get back
to you ASAP.We use a hack to get "fill" working on svgs, search for "filter" in the
tabcenter.css

On Sat, May 12, 2018, at 2:29 PM, Mélanie Chauvel (ariasuni) wrote:
> Are you satisfied with the proposed changes?


> — You are receiving this because you commented. Reply to this email
> directly, view it on GitHub[1], or mute the thread[2].> 

Links:

  1. https://github.com/eoger/tabcenter-redux/pull/328#issuecomment-388573999
  2. https://github.com/notifications/unsubscribe-auth/AGIH_2r6Air-kKH3kocwEnWUPX2uSS5Xks5txyoBgaJpZM4TCMBo

Keith94 commented:
One flaw I should mention is how the close button retains the `:active` background color even though it is not being hovered. For example, click and hold on the close button and move your cursor to the left on the tab. At this point, the background should turn transparent while not being hovered, but it stays the same.

ariasuni commented:
@Keith94 Right, I noticed that Tab Center Redux’s behavior was slightly different than Firefox native one. So in the last commit, I changed the CSS to match Firefox’s behavior in this regards.

ariasuni commented:
Oops, it’s a mistake. I’ll fix that.

Keith94 commented:
@ariasuni Since you are updating some photon icons, do you mind looking at #350?

Keith94 commented:
As well, the native tab close button has a transparent background if it's not hovered. Perhaps Tab Center should reflect that, too.

ariasuni commented:
I got the icon from [Firefox’s source code](https://searchfox.org/mozilla-central/source/toolkit/themes/shared/places/defaultFavicon.svg), because it’s not on [Photon icons website](https://design.firefox.com/icons/viewer/). I modified `tab.js` to add or remove a class — `.default-favicon` — so that I can change default favicon’s color without affecting other favicons.

I removed `background-size` when only tab is hovered, so that tab close icon has a background only when the icon is hovered, to match Firefox’s behavior. @eoger, tell me if you’re okay with this change, otherwise I’ll revert it.

Keith94 commented:
Nice, I appreciate all your efforts! I just noticed a couple issues.

In your screenshot, about:debugging should be using the default favicon. As well, the default favicon itself looks grayish, but elsewhere it looks black (or white depending on theme).

![firefox_2018-05-19_09-54-23](https://user-images.githubusercontent.com/5490615/40270937-bb0f0d84-5b4a-11e8-8c28-eacf9089fa99.png)

ariasuni commented:
I simplified and fixed my code so that at first load, `about:debugging` favicon is displayed like before.

I removed the CSS that made the default favicon theme dependent, since it’s always loaded on a white background and to match how other `about:` pages favicons are displayed.

Keith94 commented:
Thanks!

>I removed the CSS that made the default favicon theme dependent, since it’s always loaded on a white background and to match how other about: pages favicons are displayed.

There is one issue in that the favicon is hard to see against the _dark theme_ background; however, any dark favicons all share this same problem... not sure if there's a good solution.

ariasuni commented:
I fixed it. It adds a `fill-color` class when favicon is default favicon or is internal to the navigator (`chrome://` urls) and finish by `.svg` (to avoid recoloring `.png`), otherwise remove it.

eoger commented on src/sidebar/tabcenter.css:
```diff
@@ -813,8 +834,7 @@ body.dark-theme {
     --primary-text-color: rgb(255, 255, 255);
-
-    --close-button-color:  rgba(238, 238, 241, .05);
+    --icon-color: #bebebe;
```
Can we rename this to `--icons-fill-color`?

eoger commented on src/sidebar/tabcenter.css:
```diff
@@ -826,31 +846,3 @@ body.dark-theme {
     color: black;
 }
-
-body.dark-theme #searchbox-icon {
```
The dark theme looked fine to me after testing your patch, but please double check this didn't break anything.

eoger commented on src/sidebar/tabcenter.css:
```diff
@@ -702,7 +723,7 @@ body[platform="win"] #searchbox-input:placeholder-shown {
     background-color: var(--close-button-hover) !important;
 }
-.tab-close:active {
+.tab-close:hover:active {
```
👌

eoger commented on src/sidebar/tabcenter.css:
```diff
@@ -680,17 +699,19 @@ body[platform="win"] #searchbox-input:placeholder-shown {
 .tab-close {
```
Almost pixel perfect, but the size of the **X** was bothering me:

![comp](https://user-images.githubusercontent.com/6424575/40318485-ccd001b4-5cf2-11e8-972d-8a293f1459e9.png)
The 40x40px gray hover background is the right size however!

eoger commented on src/sidebar/img/find.svg:
```diff
@@ -1,22 +1,6 @@
-<?xml version="1.0"?>
```
This new icon doesn't align with the other sidebar ones (bookmarks, history, synced tabs). Could we restore the old one? However, I like the removal of #standard/#inverted :)

eoger commented on src/sidebar/img/glyph-pin-pinned-12.svg:
```diff
@@ -1,24 +1,6 @@
-<?xml version="1.0"?>
 <!-- This Source Code Form is subject to the terms of the Mozilla Public
```
Are all of these new icons licensed with MPL2/in the Firefox source code?

ariasuni replied:
They are from here: https://design.firefox.com/icons/viewer/

eoger commented on src/sidebar/tabcenter.css:
```diff
@@ -577,13 +582,24 @@ body[platform="win"] #searchbox-input:placeholder-shown {
+}
+
+#tablist:not(.shrinked) .tab-icon.fill-color {
+    fill: #5a5b5c;
```
Do we need a different color in these circumstances?

eoger replied:
I've also noticed that this triggers a bug where the colors are different:

<img width="137" alt="screen shot 2018-05-21 at 12 47 04 pm" src="https://user-images.githubusercontent.com/6424575/40319150-1dd8bc20-5cf5-11e8-88a2-f4691f02cf4e.png">

eoger commented on src/sidebar/tab.js:
```diff
@@ -165,6 +163,11 @@ SideTab.prototype = {
   },
   _setIcon(favIconUrl) {
+    if (favIconUrl.startsWith("chrome://") && favIconUrl.endsWith(".svg")) {
+      this._iconView.classList.add("fill-color");
```
Let's rename the class to something more explicit, like `chrome-icon`.

ariasuni commented:
It should be all good now!

Keith94 commented:
This is how my search bar looks compared to the bookmarks sidebar one on Windows 10.

![firefox_2018-05-23_16-24-06](https://user-images.githubusercontent.com/5490615/40456509-9e72227e-5ea6-11e8-970c-93f24965345d.png)
![firefox_2018-05-23_16-23-03](https://user-images.githubusercontent.com/5490615/40456510-9f9fc5ca-5ea6-11e8-9263-3aae6ae7724a.png)

It could use some minor polish (e.g. no italics) but otherwise looks good.

ariasuni commented:
The icon could be tweaked a bit, but it seems the intent was to make it look like the URL bar, not like the native search field used in other panes. We could probably remove italic for Windows for consistency since it’s not used in others bars, though.

Also, we don’t want this pull request to become too big, at least I don’t. So, let’s focus on tabs (favicons, close buttons) for this one.

eoger commented:
Merged in https://github.com/eoger/tabcenter-redux/commit/b99e90507b787be11320f2943bde5ed24ab0f0b1.
Thank you for your patience and dedication!
What we are experiencing here with the search box is probably platform differences (looks fine to me on macOS), we should address this in another PR.
